core/src/main/java/com/linecorp/armeria/client/metric/AbstractMetricCollectingClient.java

```diff
@@ -42,7 +48,12 @@
 
     @Override
     public O execute(ClientRequestContext ctx, I req) throws Exception {
-        RequestMetricSupport.setup(ctx, meterIdPrefixFunction, false);
+        if (!ctx.hasAttr(ATTR_REQUEST_METRICS_SET)) {
```

Comment:
This is fine too but I guess you could keep the helper and pass in `AttributeKey`.

Reply:
Let me revert it. 😄